framework/src/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -211,10 +211,13 @@ public Void visitUnion(AnnotatedUnionType type, Void aVoid) {
 
         @Override
         public Void visitArray(AnnotatedArrayType type, Void aVoid) {
-            final BoundPathNode componentNode = addPathNode(new ArrayComponentNode());
-            type.setComponentType(replaceOrVisit(type.getComponentType()));
-            removePathNode(componentNode);
-
+            if (!TypesUtils.isPrimitive(type.getComponentType().getUnderlyingType())) {
```

Comment:
I improved the Javadoc on BoundsInitializer and BoundsInitializer.InitializerVisitor